### PR TITLE
Use ServerCA from GameServerAllocationPolicy instead of client secret ca.crt

### PR DESCRIPTION
**Pull request by pooneh-m**
**What type of PR is this?**
/kind feature

**What this PR does / Why we need it**:
This is adding ServerCA field to GameServerAllocationPolicy to be used instead of client ca.crt field. This unblocks cert manager adaptation for multi-cluster allocation client certs.

**Which issue(s) this PR fixes**:
Closes https://github.com/googleforgames/agones/issues/1517

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Advanced/multi-cluster-allocation.md [hunk 1]
```diff
@@ -35,13 +35,17 @@ spec:
...
     secretName: allocator-client-to-cluster-B
+    sercerCA: c2VydmVyQ0E=
   priority: 1
   weight: 100
 EOF
 ```
 
 To define the local cluster priority, similarly, an allocation rule should be defined, while leaving allocationEndpoints unset. If the local cluster priority is not defined, the allocation from the local cluster happens only if allocation from other clusters with the existing allocation rules is unsuccessful.
 
+`sercerCA` is the server TLS CA public certificate, set only if the remote server certificate is not signed by a public CA (e.g. self-signed).
```
Comment (markmandel) [inline site/content/en/docs/Advanced/multi-cluster-allocation.md:46]: Can we please wrap the references to serverCA, with a feature shortcode?

File: site/content/en/docs/Advanced/multi-cluster-allocation.md [hunk 2]
```diff
@@ -67,7 +71,7 @@ EOF
 
 The certificates are base 64 string of the certificate file e.g. `cat ${CERT_FILE} | base64 -w 0`
 
-`ca.crt` is the server TLS public certificate if it is self-signed. For simplicity, it is recommended to use one client secret per cluster and make `ca.crt` bundle of server certificates.
+Agones recommends using [cert-manager.io](https://cert-manager.io/) solution for generating client certificates.
```
Comment (markmandel) [inline site/content/en/docs/Advanced/multi-cluster-allocation.md:74]: Not for this PR, but do we want to provide some docs on this in some way down the line?

Comment (pooneh-m) [reply site/content/en/docs/Advanced/multi-cluster-allocation.md:74]: There are already documentation on allocator-service.md for cert-manager.io. I'll try to provide snippets in this doc as well, but in a different PR.

File: site/content/en/docs/Advanced/multi-cluster-allocation.md (hunk 1 above)
Comment (pooneh-m) [reply site/content/en/docs/Advanced/multi-cluster-allocation.md:47]: Done.

**Pull request merged**